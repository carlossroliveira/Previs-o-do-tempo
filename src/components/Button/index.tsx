// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';

export const Button = (): JSX.Element => {
  return (
    <ContainerSC>
      <ButtonSC>Buscar</ButtonSC>
    </ContainerSC>
  );
};
