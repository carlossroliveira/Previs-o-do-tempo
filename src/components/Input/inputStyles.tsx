import styled from 'styled-components';

export const ContainerSC = styled.div<{ error?: boolean }>`
  display: flex;
  align-items: center;

  margin-bottom: 1rem;
  border-radius: 20px;
  transform: scale(0.5);
  -ms-transform: scale(0.5);
  -webkit-transform: scale(0.5);
  background-color: ${({ theme }) => theme.background.secondary};

  box-shadow: 0 10px 40px ${({ theme }) => theme.color.primary},
    0 0 0 20px ${({ theme }) => theme.color.secondary},
    0 0 0 20px ${({ theme }) => theme.color.primary};

  &:hover {
    box-shadow: 0 10px 40px ${({ theme }) => theme.color.secondary},
      0 0 0 20px ${({ theme }) => theme.color.primary},
      0 0 0 20px ${({ theme }) => theme.color.primary};
  }

  box-shadow: ${(props) =>
    props.error
      ? '0 10px 40px tomato, 0 0 0 20px #393e46, 0 0 0 20px tomato;'
      : '0 10px 40px #eeeeee, 0 0 0 20px #393e46, 0 0 0 20px #eeeeee;'};
`;

export const InputSC = styled.input`
  text-indent: 4rem;
  padding: 0;
  margin: 0;
  border: 0;
  outline: none;
  background-color: transparent;
  color: ${({ theme }) => theme.color.secondary};

  width: 100%;
  height: 96px;
  font-size: 60px;
  line-height: 1;

  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;
