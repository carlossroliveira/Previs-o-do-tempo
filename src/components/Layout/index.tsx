// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Header } from './part/Header';
import { ContentOne } from './part/ContentOne';
import { ContentTwo } from './part/ContentTwo';
import { Footer } from './part/Footer';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './layoutStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Layout = (): JSX.Element => {
  return (
    <ContainerSC>
      <Header />
      <ContentOne />
      <ContentTwo />
      <Footer />
    </ContainerSC>
  );
};
