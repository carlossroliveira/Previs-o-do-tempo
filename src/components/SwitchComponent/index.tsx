// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useState } from 'react';
import { useMyHook } from '../hooks/theme';
import { ParagraphSC, Toggle } from './SwitchComponentStyles';

export const SwitchComponent = (): JSX.Element => {
  const { handleThemes } = useMyHook();

  const [first, setfirst] = useState<boolean>(false);

  const test = () => {
    setfirst((firstt) => !firstt);
    handleThemes();
  };

  return (
    <>
      <ParagraphSC>{!first ? 'Light' : 'Dark'}</ParagraphSC>
      <Toggle
        checked={first}
        uncheckedIcon={false}
        checkedIcon={false}
        onChange={test}
      />
    </>
  );
};
