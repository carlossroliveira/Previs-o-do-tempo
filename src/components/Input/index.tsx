// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useMemo } from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { useMyHookApplication } from '../../Context/contextApplication/ContextTheme';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, InputSC } from './inputStyles';
import { Hourglass } from 'react-spinners-css';

export const Input = (): JSX.Element => {
  const { storage, loading, validateInput, onChangeInformation } =
    useMyHookApplication();

  const errorMessage = useMemo(() => {
    return !!validateInput;
  }, [validateInput]);

  return (
    <ContainerSC error={errorMessage}>
      <InputSC
        type="text"
        placeholder="Digite seu estado"
        value={storage}
        onChange={onChangeInformation}
      />

      {loading && <Hourglass color={'#393E46'} />}
    </ContainerSC>
  );
};
