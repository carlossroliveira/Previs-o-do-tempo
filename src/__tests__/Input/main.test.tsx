// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import theme from '../../styles/themes/dark';
import { Input } from '../../components/Input';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../utils/withThemeComponent';

const InputWithTheme = WithThemeComponent(Input);

describe('Input Testing', () => {
  it('Should render the Input component', () => {
    render(<InputWithTheme />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  it('should render the input component enabled', () => {
    render(<InputWithTheme />);
    expect(screen.getByRole('textbox')).not.toBeDisabled();
  });

  it('should render the Input component with the right color selected', () => {
    render(<InputWithTheme />);
    expect(screen.getByRole('textbox')).toHaveStyle({
      color: theme.color.secondary,
    });
  });

  it('this should allow you to type the input', () => {
    render(<InputWithTheme />);
    const input = screen.getByRole('textbox') as HTMLInputElement;
    fireEvent.click(input);
    fireEvent.change(input, { target: { value: 'Salvador' } });
    expect(input.value).toBe('Salvador');
  });

  it('should match snapshot', () => {
    const component = render(<InputWithTheme />);
    expect(component).toMatchSnapshot();
  });
});
