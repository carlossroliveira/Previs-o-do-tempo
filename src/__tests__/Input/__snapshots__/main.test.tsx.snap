// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Testing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  border-radius: 20px;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -ms-transform: scale(0.5);
  -webkit-transform: scale(0.5);
  background-color: #EEEEEE;
  box-shadow: 0 10px 40px #EEEEEE, 0 0 0 20px #393E46, 0 0 0 20px #EEEEEE;
  box-shadow: 0 10px 40px #eeeeee,0 0 0 20px #393e46,0 0 0 20px #eeeeee;
}

.c0:hover {
  box-shadow: 0 10px 40px #393E46, 0 0 0 20px #EEEEEE, 0 0 0 20px #EEEEEE;
}

.c1 {
  text-indent: 4rem;
  padding: 0;
  margin: 0;
  border: 0;
  outline: none;
  background-color: transparent;
  color: #393E46;
  width: 100%;
  height: 96px;
  font-size: 60px;
  line-height: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<body>
    <div>
      <div
        class="c0"
      >
        <input
          class="c1"
          placeholder="Digite seu estado"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  border-radius: 20px;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -ms-transform: scale(0.5);
  -webkit-transform: scale(0.5);
  background-color: #EEEEEE;
  box-shadow: 0 10px 40px #EEEEEE, 0 0 0 20px #393E46, 0 0 0 20px #EEEEEE;
  box-shadow: 0 10px 40px #eeeeee,0 0 0 20px #393e46,0 0 0 20px #eeeeee;
}

.c0:hover {
  box-shadow: 0 10px 40px #393E46, 0 0 0 20px #EEEEEE, 0 0 0 20px #EEEEEE;
}

.c1 {
  text-indent: 4rem;
  padding: 0;
  margin: 0;
  border: 0;
  outline: none;
  background-color: transparent;
  color: #393E46;
  width: 100%;
  height: 96px;
  font-size: 60px;
  line-height: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div>
    <div
      class="c0"
    >
      <input
        class="c1"
        placeholder="Digite seu estado"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
